src/lib/Component/EventListener/TwigComponentCollectorListener.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\TwigComponents\Component\EventListener;
+
+use Ibexa\TwigComponents\Component\Event\RenderGroupEvent;
+use Ibexa\TwigComponents\Component\Event\RenderSingleEvent;
+use Ibexa\TwigComponents\DataCollector\TwigComponentCollector;
+
+final class TwigComponentCollectorListener
```

Comment:
You can - should - use `EventSubscriberInterface`.

A general rule of thumb when picking whether you want to use subscriber or listener is flexibility. Listeners should be used if they need to be dynamically added / are generic enough to be attached to more events.

Also, consider using higher priority than normal (since you can use priorities for event listeners). We want to collect data early to prevent event propagation being stopped from impacting data collection.